File: backend/src/models/Driver.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface IDriver extends Document {
  driverId: string;
  name: string;
  phone: string;
  vehicleId: string;
  vehicleType: string;
  status: 'online' | 'offline' | 'busy' | 'in_transit';
  currentLocation: {
    lat: number;
    lng: number;
  };
  stationHub?: string;
  rtoCode?: string;
  rating: number;
  reliability: number;
  churnRisk: number;
  earnings: number;
  completedDeliveries: number;
  cancellationRate: number;
  updatedAt: Date;
}

const DriverSchema: Schema = new Schema({
  driverId: { type: String, required: true, unique: true },
  name: { type: String, required: true },
  phone: { type: String, required: true },
  vehicleId: { type: String, required: true },
  vehicleType: { type: String, default: '32ft Heavy Trailer' },
  status: { type: String, enum: ['online', 'offline', 'busy', 'in_transit'], default: 'online' },
  currentLocation: {
    lat: { type: Number, required: true, default: 13.0844 },
    lng: { type: Number, required: true, default: 80.2936 }
  },
  stationHub: { type: String, default: 'Chennai Port CFS' },
  rtoCode: { type: String, default: '01' },
  rating: { type: Number, default: 4.92 },
  reliability: { type: Number, default: 0.98 },
  churnRisk: { type: Number, default: 0.02 },
  earnings: { type: Number, default: 48500 },
  completedDeliveries: { type: Number, default: 64 },
  cancellationRate: { type: Number, default: 0.01 }
}, {
  timestamps: true
});

export default mongoose.model<IDriver>('Driver', DriverSchema);

File: backend-java/src/main/java/com/fleet/dispatch/model/Driver.java
package com.fleet.dispatch.model;

import java.util.Date;

public class Driver {
    private String driverId;
    private String name;
    private String phone;
    private String vehicleId;
    private String vehicleType = "bike"; // bike, car, truck, scooter
    private String status = "offline"; // online, offline, busy
    private Location currentLocation = new Location(19.0760, 72.8777);
    private double rating = 5.0;
    private double reliability = 1.0;
    private double churnRisk = 0.0;
    private double earnings = 0.0;
    private int completedDeliveries = 0;
    private double cancellationRate = 0.0;
    private Date updatedAt = new Date();

    public Driver() {}

    public String getDriverId() {
        return driverId;
    }

    public void setDriverId(String driverId) {
        this.driverId = driverId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getVehicleId() {
        return vehicleId;
    }

    public void setVehicleId(String vehicleId) {
        this.vehicleId = vehicleId;
    }

    public String getVehicleType() {
        return vehicleType;
    }

    public void setVehicleType(String vehicleType) {
        this.vehicleType = vehicleType;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public Location getCurrentLocation() {
        return currentLocation;
    }

    public void setCurrentLocation(Location currentLocation) {
        this.currentLocation = currentLocation;
    }

    public double getRating() {
        return rating;
    }

    public void setRating(double rating) {
        this.rating = rating;
    }

    public double getReliability() {
        return reliability;
    }

    public void setReliability(double reliability) {
        this.reliability = reliability;
    }

    public double getChurnRisk() {
        return churnRisk;
    }

    public void setChurnRisk(double churnRisk) {
        this.churnRisk = churnRisk;
    }

    public double getEarnings() {
        return earnings;
    }

    public void setEarnings(double earnings) {
        this.earnings = earnings;
    }

    public int getCompletedDeliveries() {
        return completedDeliveries;
    }

    public void setCompletedDeliveries(int completedDeliveries) {
        this.completedDeliveries = completedDeliveries;
    }

    public double getCancellationRate() {
        return cancellationRate;
    }

    public void setCancellationRate(double cancellationRate) {
        this.cancellationRate = cancellationRate;
    }

    public Date getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(Date updatedAt) {
        this.updatedAt = updatedAt;
    }
}
